### DISCUSSION TITLE
Conditional Rule Execution

### DISCUSSION THREAD

**Discussion by tapaj** [CacheControl/json-rules-engine#296] [Q&A] [answered]
Hi,

I wanted to implementing a chain of three rules, where the second and third rule depends on the result of the first rule. Consider this use case:

![Sample Rule Chain](https://user-images.githubusercontent.com/29328125/160279647-cb64b73d-7489-42e8-866d-60c3752e992f.png)

For this, I am creating a new fact using `engine.addRuntimeFact("isMinor", <true | false>)` and using think fact for the next two rules. The problem is that both `softDrinkSelectionRule` and `AlcoholicBeverageSelectionRule` gets executed. I want only one rule to be evaluated on the basis of the outcome of the previous rule.

Is there a better way to implement this?

**Comment by chris-pardy** [comment, marked as answer]
If you specify the priority of the rules so that `MinorOrNotRule` is a higher priority then the other rules an each rule has a high priority `isMinor` check then while technically both rules will run only the `isMinor` check would run.
See also #336 for my proposal on how the engine could more easily support this.
